tests/suite-unit/PluginFlyvemdmAgent.php

```diff
@@ -243,4 +255,96 @@ public function testGetAgents() {
          $this->boolean(isset($agents[$agent->getID()]))->isTrue();
       }
    }
+
+   /**
+    * @tags testAddDefaultJoin
+    */
+   public function testAddDefaultJoin() {
+      $instance = $this->newTestedInstance();
+      $result = $instance->addDefaultJoin();
+      $this->string($result)->isEmpty();
+
+   }
+
+   /**
+    * @tags testAddDefaultWhere
+    */
+   public function testAddDefaultWhere() {
+      $instance = $this->newTestedInstance();
+      $result = $instance->addDefaultWhere();
+      $this->string($result)->isEmpty();
+
+   }
+
+   /**
+    * @tags testGetTopic
+    */
+   public function testGetTopic() {
+      $instance = $this->newTestedInstance();
+      $this->variable($instance->getTopic())->isNull();
+      // TODO: mock internal method to return the correct value
```

Comment:
no need to mock; just enroll an agent with the helper methods. As long you know the serial and the entity ID of your tested instance, you can determine  the expected topic.

When DI will be in place we can consider to mock safely. (by the way the related  PR very likely needs a rebase and an update)

Same comment applies to many test methods below